go/ql/src/experimental/CWE-770/DenialOfService.ql

```diff
@@ -0,0 +1,59 @@
+/**
+ * @name Denial Of Service
+ * @description slices created with the built-in make function from user-controlled sources using a
+ * maliciously large value possibly leading to a denial of service.
+ * @kind path-problem
+ * @problem.severity high
+ * @security-severity 9
+ * @id go/denial-of-service
+ * @tags security
+ *       experimental
+ *       external/cwe/cwe-770
+ */
+
+import go
+
+
+/**
+ * Class for defining a predicate to check for denial of service sanitizer guard.
```

Comment:
```suggestion
 * Holds if the guard `g` on its branch `branch` checks that `e` is not constant and is less than some other value.
```
See https://github.com/github/codeql/blob/main/docs/qldoc-style-guide.md#qldoc-for-predicates for the guidelines. (Another option is just to make this predicate private, so it doesn't need a QLDoc.)